src/frontend/src/session/cursor_manager.rs

```diff
@@ -343,15 +344,22 @@ impl SubscriptionCursor {
 
     async fn get_next_rw_timestamp(
         seek_timestamp: u64,
-        table_id: u32,
+        table_id: &TableId,
         expected_timestamp: Option<u64>,
         handle_args: HandlerArgs,
+        dependent_subscription: &SubscriptionCatalog,
     ) -> Result<(Option<u64>, Option<u64>)> {
+        let session = handle_args.session;
+        // Test subscription existence
+        session.get_subscription_by_schema_id_name(
+            dependent_subscription.schema_id,
+            &dependent_subscription.name,
+        )?;
+
         // The epoch here must be pulled every time, otherwise there will be cache consistency issues
-        let new_epochs = handle_args
-            .session
+        let new_epochs = session
             .catalog_writer()?
-            .list_change_log_epochs(table_id, seek_timestamp, 2)
+            .list_change_log_epochs(table_id.table_id(), seek_timestamp, 2)
```

Comment:
I have several questions related to `list_change_log_epochs`:
1. Why do we put `list_change_log_epochs` in `catalog_writer`? It is a read-only operation.
2. Why do we have `list_change_log_epochs` in [SessionImpl](https://github.com/risingwavelabs/risingwave/blob/267e6c9410c13a23fe064c00ae46d9163e7de30e/src/frontend/src/session.rs#L956) but unused?
3. What will `list_change_log_epochs` return if the requested table_id is dropped? Should change `list_change_log_epochs` to make it return an error when the table_id is not found to handle the case when subscription is dropped instead of adding a new method to check existence of a subscription?